### DISCUSSION TITLE
Typescript - findOne can return null?

### DISCUSSION THREAD

**Discussion by alan345** [prisma/prisma#3849] [Q&A] [answered]
2 different servers `expressJs` are hitting the same `PrismaClient`.
In the first server `findOne` can return `IssuedCard` or `null`. In the second instance can never return `null`

the snippet is similar in both servers: 
```ts
    const updatedIssuedCard = await req.prisma.issuedCard.findOne({
      where: { id: newIssuedCard.id },
      include: { user: true, createdBy: true, companie: true },
    })
    let test = updatedIssuedCard.companie.id 
```

See print screens:
- Server1
![image](https://user-images.githubusercontent.com/15246526/94974558-cb3f7700-04c3-11eb-8ec6-658b42616287.png)
The error is `Object is possibly 'null'.ts(2531)`

- Server2
![image](https://user-images.githubusercontent.com/15246526/94974336-27ee6200-04c3-11eb-88b9-22cb07969776.png)

Is there an option somewhere?

**Comment by alan345** [comment]
Correct me if I am wrong but I think it is because of:

```
// tsconfig.json
{
  "compilerOptions": {
    "strict": true,
}
```

What is your recommendation? Should we set it to `true`?

**Reply by ryands17** [reply]
Hey @alan345 👋 
This is a strange one. I currently have tried with the following schema:

```prisma
model Post {
  id        Int      @id @default(autoincrement())
  published Boolean  @default(false)
  title     String
  content   String?
  authorId  Int?
  author    User?    @relation(fields: [authorId], references: [id])
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}

model User {
  id        Int      @id @default(autoincrement())
  name      String?
  posts     Post[]
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}
```

And with the include, it looks as follows:

![image](https://user-images.githubusercontent.com/19697099/95066665-5452ea00-0720-11eb-923b-a7cb8f31de19.png)

The options in `tsconfig.json` don't alter those so I don't think options there are causing this.

**Comment by 2color** [comment, marked as answer]
Indeed, @alan345 this is due to the `strict` mode in `tsconfig.json`. `strict` mode enables [strictNullChecks](https://www.typescriptlang.org/tsconfig#strictNullChecks) which is why you're using the type error. It's generally recommended to set `strict` to `true`. 

With [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining) you can avoid the type errors that you're seeing, for example: `updatedIssuedCard?.companie?.id`

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
